cli/src/semgrep/output.py

```diff
@@ -284,11 +284,16 @@ def output(
         state = get_state()
         self.has_output = True
         self.rules = self.rules.union(rule_matches_by_rule.keys())
-        self.rule_matches = [
-            match
-            for matches_of_one_rule in rule_matches_by_rule.values()
-            for match in matches_of_one_rule
-        ]
+        # Sort findings by rule name. We rely on the list of findings
+        # being already sorted for each rule.
+        self.rule_matches = sorted(
+            (
+                match
+                for matches_of_one_rule in rule_matches_by_rule.values()
+                for match in matches_of_one_rule
+            ),
+            key=lambda x: x.rule_id,
+        )
```

Comment:
This appears unnecessary but actually deduplicates the findings. :thinking:

Reply:
No, this sorting actually has the effect of removing a finding that's unique.

Reply:
Nevermind. The problem is that accidentally used parentheses instead of square brackets and Python did something weird without telling me.